src/components/OSCALControlProse.js

```diff
@@ -55,17 +55,30 @@ function getParameterLabel(parameters, parameterId) {
 /**
  * Finds a parameter setting in a component statement
  * @param {Object} statementByComponent
+ * @param {Object} implReqSetParameters
  * @param {String} parameterId
  * @returns the parameter label
  */
-function getParameterValue(statementByComponent, parameterId) {
+function getParameterValue(
+  statementByComponent,
+  implReqSetParameters,
+  parameterId
+) {
   // Locate matching parameter to parameterId
-  if (!statementByComponent["set-parameters"]) {
+  let foundParameterSetting = null;
+  // Locate set-parameters when found in the by-component
+  if (statementByComponent?.["set-parameters"]) {
+    foundParameterSetting = statementByComponent["set-parameters"].find(
+      (parameterSetting) => parameterSetting["param-id"] === parameterId
```

Comment:
If we're using this twice (once per branch), we should move it to it's own function within `getParameterValue`.

```javascript
// This assumes that `parameterId` from `getParameterValue` is in scope
function parameterHasGivenId(parameterSetting) {
  return parameterSetting["param-id"] === parameterId;
}
```

And then you can use it as:

```javascript
if (statementByComponent?.["set-parameters"]) {
  foundParameterSetting = statementByComponent["set-parameters"].find(parameterHasGivenId);
} else if (implReqSetParameters) {
  foundParameterSetting = implReqSetParameters.find(parameterHasGivenId);
} else {
  return null;
}
```

But I'd also suggest that we've got quite a bit of repetition there still (between the if and the actual clause). And we may want to consider also doing something like:

```
const setParameters = () => statementByComponents?.["set-parameters"] || implReqSetParameters;
const foundParameterSetting = setParameters()?.find(parameterHasGivenId);
```

And since the next check is `if (!foundParameterSetting?.values) { return null; }` we shouldn't need to modify any code. But we could do a `if (!foundParameterSetting) { return null; }` if we really wanted to.

This has the benefits of not only reducing repetition but also keeping more things `const`. It's also one less `null` that we really need to worry about.

Reply:
I made all recommended changes except the change in the conditional because we do `foundParameterSetting.values.toString()` right after the check and if `foundParameterSetting.values` is null, we would get an error.